src/front_input.c

```diff
@@ -2585,64 +2589,77 @@ void process_cheat_mode_selection_inputs(unsigned char *value)
         case PSt_PlaceTerrain:
         {
+            SlabKind new_value = gameadd.chosen_terrain_kind;
             if (is_key_pressed(KC_0, KMod_NONE))
             {
-                *value = SlbT_ROCK;
+                new_value = SlbT_ROCK;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_0);
             }
             else if (is_key_pressed(KC_1, KMod_NONE))
             {
-                *value = SlbT_GOLD;
+                new_value = SlbT_GOLD;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_1);
             }
             else if (is_key_pressed(KC_2, KMod_NONE))
             {
-                *value = SlbT_GEMS;
+                new_value = SlbT_GEMS;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_2);
             }
             else if (is_key_pressed(KC_3, KMod_NONE))
             {
-                *value = SlbT_EARTH;
+                new_value = SlbT_EARTH;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_3);
             }
             else if (is_key_pressed(KC_4, KMod_NONE))
             {
-                *value = SlbT_TORCHDIRT;
+                new_value = SlbT_TORCHDIRT;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_4);
             }
             else if (is_key_pressed(KC_5, KMod_NONE))
             {
-                *value = SlbT_PATH;
+                new_value = SlbT_PATH;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_5);
             }
             else if (is_key_pressed(KC_6, KMod_NONE))
             {
-                *value = SlbT_CLAIMED;
+                new_value = SlbT_CLAIMED;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_6);
             }
             else if (is_key_pressed(KC_7, KMod_NONE))
             {
-                *value = SlbT_LAVA;
+                new_value = SlbT_LAVA;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_7);
             }
             else if (is_key_pressed(KC_8, KMod_NONE))
             {
-                *value = SlbT_WATER;
+                new_value = SlbT_WATER;
+                set_players_packet_action(player, PckA_CheatSwitchTerrain, new_value, 0, 0, 0);
                 clear_key_pressed(KC_8);
             }
             else if (is_key_pressed(KC_9, KMod_NONE))
             {
-                *value = rand() % (5) + 4;
+                new_value = rand() % (5) + 4;
```

Comment:
rand() shouldn't be used anywhere I believe, I'm guessing it would need a PLAYER_RANDOM in this case and if it doesn't need to be synced